File: 프로그래머스/unrated/181856. 배열 비교하기/배열 비교하기.java
class Solution {
    public int sum (int[] arr)
    {
        int sum = 0;
        for (int num : arr)
            sum += num;
        return sum;
    }
    public int solution(int[] arr1, int[] arr2) {
        int len1 = arr1.length;
        int len2 = arr2.length;
        if (len1 == len2)
        {
            int sum1 = sum(arr1);
            int sum2 = sum(arr2);
            return (sum1 < sum2 ? -1 : (sum1 == sum2 ? 0 : 1));
        }
        return len1 < len2 ? -1 : 1;
    }
}

File: 프로그래머스/0/181856. 배열 비교하기/배열 비교하기.js
function solution(arr1, arr2) {
    const ldiff = arr1.length - arr2.length;
    if (ldiff !== 0)
        return 0 < ldiff ? 1 : -1;
    const diff = arr1.reduce((s,v) => s + v) - arr2.reduce((s,v) => s + v);
        return 0 < diff ? 1 : (diff < 0 ? -1 : 0);
}